homeassistant/components/sonos/household_coordinator.py

```diff
@@ -25,7 +25,7 @@ def __init__(self, hass: HomeAssistant, household_id: str) -> None:
         self.household_id = household_id
         self.async_poll: Callable[[], Coroutine[None, None, None]] | None = None
         self.last_processed_event_id: int | None = None
-        self.cache_update_lock: asyncio.Lock | None = None
+        self.cache_update_lock = asyncio.Lock()
```

Comment:
This can't be done as it's not instantiated in async context. It was intentionally deferred and created in a task.

Reply:
You can set `self.cache_update_lock: asyncio.Lock` as a class type hint instead and then drop the line from `__init__` method.

```python
class SonosHouseholdCoordinator:
    """Base class for Sonos household-level storage."""

    cache_update_lock: asyncio.Lock

    def __init__(self, hass: HomeAssistant, household_id: str) -> None:
        """Initialize the data."""
        self.hass = hass
        self.household_id = household_id
        self.async_poll: Callable[[], Coroutine[None, None, None]] | None = None
        self.last_processed_event_id: int | None = None
```

Reply:
@tkdrob did you see this message?
I still think that you should remove the `if self.cache_update_lock` lines that you added.